scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -1,48 +1,139 @@
+import metaconfig.annotation.{DeprecatedName, Description, ExampleValue}
 import metaconfig.generic.Surface
 
+case class DisablePart(
+    @Description("The symbol that indicates a 'safe' block. Optional value.")
+    @DeprecatedName(
+      "unless",
+      "Renamed to unlessInsideBlock",
+      "0.6.0"
+    )
+    unlessInsideBlock: Option[Symbol.Global],
+    @Description(
+      "When true also blocks symbols in the generated code. Default to false. " +
+        "Ignored if unlessInsideBlock is specified.")
+    unlessSynthetic: Boolean,
+    @Description("The unsafe that are banned unless inside a 'safe' block")
+    symbols: List[CustomMessage[Symbol.Global]])
+
+object DisablePart {
+  implicit val surface: Surface[DisablePart] =
+    generic.deriveSurface[DisablePart]
+  // NOTE(olafur): metaconfig.generic.deriveDecoder requires a default base values.
+  // Here we require all fields to be provided by the user so we write the decoder manually.
+  implicit val reader: ConfDecoder[DisablePart] =
+    ConfDecoder.instanceF[DisablePart] {
+      case c: Conf.Obj =>
+        (c.getOption[Symbol.Global]("unlessInsideBlock", "unless") |@|
+          c.getOrElse[Boolean]("unlessSynthetic")(false) |@|
+          c.get[List[CustomMessage[Symbol.Global]]]("symbols")).map {
+          case ((a @ Some(_), _), c) => DisablePart(a, false, c)
+          // when unlessInsideBlock is specified we ignore unlessSynthetic
+          case ((None, b), c) => DisablePart(None, b, c)
+        }
+      case _ => Configured.NotOk(ConfError.message("Wrong config format"))
+    }
+}
+
 case class DisableConfig(
-    @Description("The list of symbols to disable.")
-    @ExampleValue(
-      """[
-        |  # With custom message (recommended)
-        |  {
-        |    symbol = "scala.Predef.any2stringadd"
-        |    message = "Use explicit toString before calling +"
-        |  }
-        |  {
-        |    symbol = "scala.Any"
-        |    message = "Explicitly type annotate Any if this is intentional"
-        |  }
-        |  # Without custom message (discouraged)
-        |  "com.Lib.implicitConversion"
-        |]""".stripMargin)
-    symbols: List[CustomMessage[Symbol.Global]] = Nil
+    @Description("The list of disable configurations")
+    @ExampleValue("""[
+                    |  { // default mode
+                    |    symbols = [
+                    |      "scala.Any.asInstanceOf"
+                    |    ]
+                    |  }
+                    |  { // unlessSynthetic mode
+                    |    unlessSynthetic = true
+                    |    symbols = [
+                    |      {
+                    |        symbol = "scala.Predef.any2stringadd"
+                    |        message = "Use explicit toString be calling +"
+                    |      }
+                    |    ]
+                    |  }
+                    |  { // unlessInsideBlock mode
+                    |    unlessInsideBlock = "com.IO"
```

Comment:
Can we remove this example? I think it's enough to have one example and `scala.util.Try` seems like the most easy to understand